lib/datadog.ts

```diff
@@ -62,4 +72,30 @@ export class Datadog extends cdk.Construct {
       this.transport.setEnvVars(lambdaFunctions);
     }
   }
+
+  public validateProps(props: DatadogProps) {
+    const siteList: string[] = ["datadoghq.com", "datadoghq.eu", "us3.datadoghq.com", "ddog-gov.com"];
+
+    if (props.extensionLayerVersion !== undefined) {
+      if (props.forwarderARN !== undefined) {
+        throw new Error("`extensionLayerVersion` and `forwarderArn` cannot be set at the same time.");
+      }
+      if (props.apiKey === undefined && props.apiKMSKey === undefined) {
+        throw new Error("When `extensionLayer` is set, `apiKey` or `apiKMSKey` must also be set.");
+      }
```

Comment:
What happens when `extensionlayer` is set, `apiKey` is set, AND `apiKMSKey` is set?

It seems like we never want `apiKey` and `KMSKey` to both be set, regardless of all other settings. We can probably refactor this code to reflect that.

Reply:
I refactored it in datadog.ts